src/FiscalYears/FiscalYearsList/FiscalYearsList.js

```diff
@@ -0,0 +1,161 @@
+import React, { useCallback } from 'react';
+import PropTypes from 'prop-types';
+import {
+  Route,
+  withRouter,
+} from 'react-router-dom';
+import ReactRouterPropTypes from 'react-router-prop-types';
+import { FormattedMessage } from 'react-intl';
+
+import {
+  Paneset,
+  MultiColumnList,
+} from '@folio/stripes/components';
+import {
+  FiltersPane,
+  ResultsPane,
+  ResetButton,
+  SingleSearchForm,
+  useLocationFilters,
+  useLocationSorting,
+  useToggle,
+} from '@folio/stripes-acq-components';
+
+import { FISCAL_YEAR_ROUTE } from '../../common/const';
+import FinanceNavigation from '../../common/FinanceNavigation';
+
+import FiscalYearDetails from '../FiscalYearDetails';
+import { FiscalYearsListFilterContainer } from './FiscalYearsListFilter';
+import {
+  searchableIndexes,
+} from './FiscalYearsListSearchConfig';
+import FiscalYearsListLastMenu from './FiscalYearsListLastMenu';
+
+const resultsPaneTitle = <FormattedMessage id="ui-finance.fiscalyear" />;
+const visibleColumns = ['name', 'code', 'description'];
+const sortableFields = ['name', 'code', 'description'];
+const columnMapping = {
+  name: <FormattedMessage id="ui-finance.fiscalYear.list.name" />,
+  code: <FormattedMessage id="ui-finance.fiscalYear.list.code" />,
+  description: <FormattedMessage id="ui-finance.fiscalYear.list.description" />,
+};
+
+const FiscalYearsList = ({
+  history,
+  isLoading,
+  location,
+  onNeedMoreData,
+  resetData,
+  fiscalYears,
+  fiscalYearsCount,
+}) => {
+  const [
+    filters,
+    searchQuery,
+    applyFilters,
+    applySearch,
+    changeSearch,
+    resetFilters,
+    changeIndex,
+    searchIndex,
+  ] = useLocationFilters(location, history, resetData);
+  const [
+    sortingField,
+    sortingDirection,
+    changeSorting,
+  ] = useLocationSorting(location, history, resetData, sortableFields);
+
+  const [isFiltersOpened, toggleFilters] = useToggle(true);
+
+  const openFiscalYearDetails = useCallback(
+    (e, meta) => {
+      history.push({
+        pathname: `${FISCAL_YEAR_ROUTE}/${meta.id}/view`,
+        search: location.search,
+      });
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [location.search],
+  );
+
+  const renderLastMenu = useCallback(() => <FiscalYearsListLastMenu />, []);
```

Comment:
is `useCallback` required here?

Reply:
why not, without it will be always new function, with it only 1